KHJ/6Day/이상한문자 만들기.js

```diff
@@ -0,0 +1,24 @@
+function solution(s) {
+    let answer = []; 
+    const words = s.split(' '); 
+    for (let i = 0; i < words.length; i++) {
+        let transformedWord = ''; 
+        let letterCount = 0; 
+        for (let j = 0; j < words[i].length; j++) {
+            const char = words[i][j]; 
+             if (/[a-zA-Z]/.test(char)) { //현재 문자가 알파벳인지 확인
+                // 정규 표현식 /[a-zA-Z]/를 사용하여 대문자와 소문자를 모두 포함
+                letterCount++; // 알파벳 문자가 발견되면 letterCount에 1증가시킴
+                if (letterCount % 2 === 1) {
+                    transformedWord += char.toUpperCase();  
+                } else {
+                    transformedWord += char.toLowerCase(); 
+                }
+            } else {
+                transformedWord += char; //알파벳이 아니면 그대로 transformeWorld에 추가
```

Comment:
정규표현식을 벌써 공부해보셨나보네요.. 대단합니다

근데 char에 들어가는 값이 알파벳이 아닌 경우가 있는건가요?
문제를 봤을땐 전부 알파벳으로 구성되어 있는거 같은데 반례가 있나 해서 질문드립니다 ~

Reply:
' 와 같은 특수문자가 들어가는 경우가 있어 그렇게 짜보았습니다

Reply:
```suggestion
               function solution(s) {
    let answer = []; 
    const words = s.split(' '); 
    for (let i = 0; i < words.length; i++) {
        let transformedWord = ''; 
        let letterCount = 0; 
        for (let j = 0; j < words[i].length; j++) {
            const char = words[i][j]; 
            // 정규 표현식 /[a-zA-Z]/를 사용하여 대문자와 소문자를 모두 포함
            letterCount++; // 알파벳 문자가 발견되면 letterCount에 1증가시킴
            if (letterCount % 2 === 1) {
                transformedWord += char.toUpperCase();  
            } else {
                transformedWord += char.toLowerCase(); 
            }
        }
        answer.push(transformedWord); 
    }
    return answer.join(' '); 
}
```
프로그래머스에서 위와 같이 정규표현식 조건을 없애고 테스트를 돌렸는데 통과가 되는거 같아요 ~
한번 확인해보시고 답글 부탁드립니다 ~